modules/internal/ChapelUtil.chpl

```diff
@@ -97,6 +97,34 @@ module ChapelUtil {
       }
     }
   }
+
+  //
+  // safeMul: If a and b are of type t, return true iff no
+  //  overflow/underflow would occur for a * b
+  //
+  proc safeMul(a: ?t, b:t){
+    if !isIntegralType(t) then
+      compilerError("Values must be of integral type.");
+    if a>0{
+      if b>0 {
+        if a > max(t)/b then return false;
+      }
+      else {
+        if b < min(t)/a then return false;
+      }
+    }
+    else {
+      if b>0{
+        if a < min(t)/b then return false;
+      }
+      else {
+        if a!=0 && b < max(t)/a then return false;
+      }
+    }
+    // //if all the case are false then multiplication is safe
+    return true;
```

Comment:
This is unecessary now, removing this part, before merging(if possible).

Reply:
What is unnecessary?

Reply:
`return true` will never be reached by program ( all the cases are covered in if-else)

Reply:
All the `false` cases seems to be covered by those, am I missing something?

Reply:
my bad. Yeah obviously there has to be a true statement in the end.